contribute/creating_package.md

```diff
@@ -96,11 +95,56 @@ To run tests on your package:
+- run `test`, or, alternatively `test Tiddlywinks` to perform tests. Test summary will provide you with the number of passed tests out of the total number of tests.
+
+## Continuous integration
+
+Continuous integration (CI) is a software development practice of merging all working copies into a mainline repository. CI tools can help you test the package on various versions of Julia, as well as various operating systems (those options are governed by the file `.travis.yml` in the case of using Travis CI).
+
+To enable CI, make sure to have pushed all of the changes to GitHub. Click on the badge in the README file in the GitHub repository of the package, and sign in with Travis CI. If you have not set up CI with Travis before, follow these steps:
+
+- on the left tab, click the `+` sign
+- on the left tab, press `Synch account`
+- on the right tab, search for the name of your newly created package. A `Settings` button should appear in front of the name of your package.
+
+Using the second badge, you can make use of Appveyor. Click on the badge, sign in with your GitHub account, authorize repositories. On the list of authorised repositories, find the name of your package, and click `Add`. You are all set to monitor changes in the package.
+
+## Propose changes via Pull Requests
+
+To propose a change to an existing repository, you need to create a pull request (PR). Follow the steps below. The example
+describes a case when you would like to add more testing options to your package:
+
+- clone the repository which you would like to change (skip this step if the repository already exists under your username on GitHub)
+
+- create a new branch of the project
+
+Example: create a branch of `Tiddlywinks.jl` called `more_testing_options`
+
+- make and save changes on the new branch
+
+Example: edit file `.travis.yml` to contain more Julia versions to test
+
+```
+julia:
+  - 1.0
+  - 1.3
+  - 1.6
+  - nightly
+```
+
+- push the changes to your GitHub repository
+
+- to create a pull request on GitHub, press the arrow to the right from the word `Contribute`, and then press `Open pull request`
+
+- add title for your pull request and (optional) leave more comments to the proposed changes
+
+- click `Create pull request`
+
+Now your request has appeared on the tab `Pull requests` of the package. If you click on this tab, `Files changed` will show you exact changes that you are proposing. [Continuous integration](#continuous-integration) tools, if they have been included into the project, will run the tests to check that they can be passed by the new branch.
 
-## Continuous integration (CI)
+If all test will be passed, package maintainers will review your code and either leave comments or approve it to be merged into the `main` branch.
 
-Make sure to have pushed all of the changes to GitHub. Click on the first badge which has appeared in the GitHub repository of the package, and sign in with Travis CI.
+- delete the branch after it has been merged in
```

Comment:
Only people with merge access will be able to create branches on the main repo itself, otherwise, it will all be done via a fork. We should perhaps touch on to link to a "Getting started with Git Video" (we may want to make one ourselves to drive people to the Julia Language YouTube but that's for another day).